### DISCUSSION TITLE
Choose a baseline ISA for CLANGARM64 builds

### DISCUSSION THREAD

**Discussion by hmartinez82** [msys2/MINGW-packages#18829] [Ideas]
More and more packages are being built for CLANGARM64, but it occurred to me that we may be leaving some performance on the table.

Of course, it's better to have a native build first regardless of optimizations, but in light of this https://github.com/llvm/llvm-project/issues/66912 being fixed it got me wondering:

- We should at least make sure that projects that are already using NEON are enabling NEON in Windows builds
- We should pick a baseline ISA 

Bellow is, definitely not complete, possibly not correct, list of Windows on ARM laptops with their respective cores and supported baseline ISAs
| Laptop Name                     | Year Launched | CPU Name             | big Core | LITTLE Core | ISA     |
|---------------------------------|---------------|----------------------|--------------|-----------------|---------|
| Asus NovaGo                     | 2018          | Snapdragon 835       | Cortex-A73   | Cortex-A53      | ARMv8   |
| HP Envy x2                      | 2018          | Snapdragon 835       | Cortex-A73   | Cortex-A53      | ARMv8   |
| Lenovo Miix 630                 | 2018          | Snapdragon 835       | Cortex-A73   | Cortex-A53      | ARMv8   |
| Samsung Galaxy Book2            | 2018          | Snapdragon 850       | Cortex-A75   | Cortex-A55      | ARMv8.2 |
| Huawei Matebook E (2019)        | 2019          | Snapdragon 850       | Cortex-A75   | Cortex-A55      | ARMv8.2 |
| Microsoft Surface Pro X         | 2019          | Microsoft SQ1        | Cortex-A76   | Cortex-A55      | ARMv8.2 |
| Lenovo Yoga 5G                  | 2020          | Snapdragon 8cx       | Cortex-A76   | Cortex-A55      | ARMv8.2 |
| Samsung Galaxybook S            | 2020          | Snapdragon 8cx       | Cortex-A76   | Cortex-A55      | ARMv8.2 |
| Microsoft Surface Pro X (SQ2)   | 2020          | Microsoft SQ2        | Cortex-A76   | Cortex-A55      | ARMv8.2 |
| Lenovo Yoga 5G                  | 2020          | Snapdragon 8cx       | Cortex-A76   | Cortex-A55      | ARMv8.2 |
| Acer Spin 7                     | 2021          | Snapdragon 8cx Gen 2 | Cortex-A76   | Cortex-A55      | ARMv8.2 |
| Microsoft Surface Pro 9 with 5G | 2023          | Microsoft SQ3        | Cortex-X1    | Cortex-A78      | ARMv8.2 |
| Lenovo ThinkPad X13s            | 2023          | Snapdragon 8cx Gen 3 | Cortex-X1    | Cortex-A78      | ARMv8.2 |
| Samsung Galaxy Book2 Pro        | 2023          | Snapdragon 8cx Gen 3 | Cortex-X1    | Cortex-A78      | ARMv8.2 |

Does that mean that `-mtune` should stay as `cortex-a53`

@Blackhex, @dennisameling feel free to chime in or tag anyone you think knows a lot about this.

**Comment by Biswa96** [comment]
Just don't break Raspberry Pi 4 support 🙏

**Comment by hmartinez82** [comment]
| Device Name                     | Year Launched | CPU Name             | Core | ISA     |
|---------------------------------|---------------|----------------------|----------------|---------|
| Raspberry Pi 4 Model B     | 2019          | Broadcom BCM2711       | Cortex-A72   | ARMv8   |
| Raspberry Pi 5                   | 2023          | Broadcom BCM2712       | Cortex-A76  |  ARMv8.2 |

@Biswa96 ^ 😉 for completion

**Comment by Blackhex** [comment]
cc @ZacWalk @movediedi
